controllers/front/DiscountController.php

```diff
@@ -43,30 +43,53 @@ public function initContent()
         $nb_cart_rules = count($cart_rules);
 
         foreach ($cart_rules as $key => &$discount ) {
+
             if ($discount['quantity_for_user'] === 0) {
                 unset($cart_rules[$key]);
             }
 
-
             $discount['value'] = Tools::convertPriceFull(
-                                            $discount['value'],
-                                            new Currency((int)$discount['reduction_currency']),
-                                            new Currency((int)$this->context->cart->id_currency)
-                                        );
+                $discount['value'],
+                new Currency((int)$discount['reduction_currency']),
+                new Currency((int)$this->context->cart->id_currency)
+            );
+            
             if ($discount['gift_product'] !== 0) {
```

Comment:
Hello,

All my discounts without gift disappear currently. I think it should be '!=' because _gift_product_ is an int eger and not a boolean ?

Thanks. :)